raster-foundry/raster-foundry
# Add projectId and layerId QPs to STAC export list endpoint and more

## Overview

This PR:
- adds projectId and layerId QPs to STAC export list endpoint (support annotate frontend)
- updates STAC export location to use the S3 prefix instead of the absolute location of the `catalog.json` (support annotate frontend)
- updates values for `label:task`, `label:property`, and `label:classes` of the label item

### Checklist

- ~New tables and queries have appropriate indices added~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

## Testing Instructions

- `./scripts/load_development_data --download` (this includes the `Florence - AL` project for local testing)
- `./scripts/console`:
   * `api/assembly`
   * `batch/assembly`
- Spin up servers and frontend. Grab a token.
- `POST` below object to `/api/stac`:
```json
{
	"name": "AL Project Test 1",
	"layerDefinitions": [
		{
			"projectId": "7e584c31-f5d1-4a02-9428-e83006642375",
			"layerId": "1a8c1632-fa91-4a62-b33e-3a87c2ebdf16"
		}
	],
	"taskStatuses": ["LABELED", "VALIDATED"]
}
```
- Grab the above export record's ID (will be referred to below as `export ID 1`)
- `POST` below object to `/api/stac`: (this is to test the list endpoint better)
```json
{
	"name": "Test 2",
	"layerDefinitions": [
		{
			"projectId": "7e584c31-f5d1-4a02-9428-e83006642375",
			"layerId": "1a8c1632-fa91-4a62-b33e-3a87c2ebdf16"
		},
                 {
			"projectId": "<another project ID>",
			"layerId": "<another project's default layer ID>"
		}
	],
	"taskStatuses": ["LABELED", "VALIDATED"]
}
```
- `GET` to `api/stac?projectId=7e584c31-f5d1-4a02-9428-e83006642375&layerId=1a8c1632-fa91-4a62-b33e-3a87c2ebdf16`. Make sure it lists two records.
- Perform the above `GET` but with different `projectId` and `layerId` combo -- make sure it gives you results as you would expect
- `docker-compose build batch`
- `./scripts/console batch bash`
- `java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main write_stac_catalog <export ID 1>`
- (if you run into DB connection issue, stop local servers)
- Check on S3 development data bucket -> `stac-exports`, there should be a new folder containing your exported catalog
- Download the label data item, make sure the property fields' values are like below (because these values were not accurate previously):
```json
"properties": {
    ...
    "label:classes": [
      {
        "name": "label",
        "classes": [
          "Passenger Vehicle",
          "Bus",
          "Truck"
        ]
      }
    ],
    "label:task": [
      "detection"
    ],
    "label:property": [
      "label"
    ]
    ...
  }
```
- Look at this export record's `export_location` in DB, it should be an S3 prefix to the export rather than a link to `catalog.json`

Closes #5138 
Closes #5136
